Content.Server/Chemistry/Components/SolutionTransferComponent.cs

```diff
@@ -129,19 +129,21 @@ async Task<bool> IAfterInteract.AfterInteract(AfterInteractEventArgs eventArgs)
                 return false;
             }
 
-
-            if (CanReceive && _entities.TryGetComponent(target, out ReagentTankComponent? tank)
-                           && solutionsSys.TryGetRefillableSolution(Owner, out var ownerRefill)
-                           && solutionsSys.TryGetDrainableSolution(target, out var targetDrain))
+            //Special case for reagent tanks, because normally clicking another container will give solution, not take it.
+            if (CanReceive  && _entities.TryGetComponent(target, out DrainableSolutionComponent? drainable) // target must be drainable
+                            && !_entities.TryGetComponent(target, out RefillableSolutionComponent? refillable) // target must not be refillable (e.g. Reagent Tanks)
+                            && solutionsSys.TryGetRefillableSolution(Owner, out var ownerRefill)
```

Comment:
I'm not familiar with solutiontransfercomp but why does it still try to get the refillable solution?

Reply:
Oh sorry didn't see this until now. On this line it's trying to get the refillable solution of the Owner, which in this case is the bucket (or whatever) that is taking solution from the tank.

The tank itself notably doesn't have a refillable solution component.

I just noticed the order on these 4 is Target, Target, Owner, Target which I agree is a little confusing.

Reply:
I think he meant to comment the line above this, i.e., why use 
```cs
!_entities.TryGetComponent(target, out RefillableSolutionComponent? refillable
```
instead of just
```cs
!_entities.HasComponent<RefillableSolutionComponent>(target)
```
Given that `refillable` goes unused.

Similarly, you don't actually use `drainable` here. Note that `solutionsSys.TryGetDrainableSolution()` Internally does a drainable-`TryGetComponent`. 
So `solutionsSys.TryGetDrainableSolution(target, out var targetDrain)` is only ever true if `_entities.TryGetComponent(target, out DrainableSolutionComponent? drainable)` is true.

Reply:
Thank you both for the review! I understand all this much better now.